service-userlayer/src/main/java/org/oskari/map/userlayer/service/UserLayerDataService.java

```diff
@@ -86,8 +87,26 @@ private static String parseFields(SimpleFeatureType schema) {
             Collection<PropertyDescriptor> types = schema.getDescriptors();
             for (PropertyDescriptor type : types) {
                 JSONObject obj = new JSONObject();
-                obj.put("name", type.getName().getLocalPart());
+                String name = type.getName().getLocalPart();
+                obj.put("name", name);
                 obj.put("type", type.getType().getBinding().getSimpleName());
+                JSONObject locales = new JSONObject();
+                // KML handling
+                if (KMLParser.KML_NAME.equals(name)){
+                    locales.put("fi", "Nimi");
```

Comment:
Not a big fan about having localization in code, but not sure where to put it either.